packages/model-viewer/src/features/scene-graph/texture-factory.ts

```diff
@@ -0,0 +1,66 @@
+
+import {MeshStandardMaterial, Texture as ThreeTexture} from 'three';
+
+import {GLTF, Texture as GLTFTexture, TextureInfo as GLTFTextureInfo} from '../../three-components/gltf-instance/gltf-2.0.js';
+
+import {Image} from './image.js';
+import {Sampler} from './sampler.js';
+import {TextureInfo, TextureUsage} from './texture-info.js';
+import {Texture} from './texture.js';
+
+
+
+export const $createTexture = Symbol('createTexture');
+export const $createTextureInfo = Symbol('createTextureInfo');
+
+/**
+ * Helper class for creating Textures and TextureInfos.
+ */
+export class TextureFactory {
+  // Creates context from Texture. This creats a texture with a valid
+  static[$createTexture](threeTexture: ThreeTexture): Texture {
+    // Creates an empty glTF data set, allows for creating a texture that's
+    // not bound to a material.
+    const emptyOnUpdate = () => {};
+    return new Texture(
+        emptyOnUpdate,
+        new Sampler(emptyOnUpdate, threeTexture, null),
+        new Image(emptyOnUpdate, threeTexture, null),
+        threeTexture,
+        null);
+  }
+
+  // A helper method for creating TextureInfo objects from gltf data.
+  static[$createTextureInfo](
+      onUpdate: () => void, usage: TextureUsage, gltf: GLTF,
+      threeTexture: ThreeTexture|null, gltfTextureInfo: GLTFTextureInfo|null,
+      correlatedMaterials: Set<MeshStandardMaterial>): TextureInfo {
+    let gltfTexture: GLTFTexture|null = null;
+
+    let texture: Texture|null = null;
+    // Creates image, sampler and texture if valid texture info is provided.
+    if (gltfTextureInfo) {
```

Comment:
Being verbose about creating the images and sampler needed to create a texture. This could be hidden inside texture constructor, but I like simplicity of doing it this way where the constructor defines what objects you need and we construct them before hand and pass them in rather than hiding their creation inside the constructor.

Reply:
I see your point, but this seems inconsistent with the rest of the style here. Each other class's constructor here constructs the next classes that are members of it. Let's stick with that convention for now for consistency.

Reply:
Done.